src/library_manager/lib_manager.c

```diff
@@ -0,0 +1,449 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Intel Corporation. All rights reserved.
+//
+// Author: Jaroslaw Stelter <jaroslaw.stelter@intel.com>
+//         Pawel Dobrowolski<pawelx.dobrowolski@intel.com>
+
+/*
+ * Dynamic module loading functions.
+ */
+
+#include <sof/audio/buffer.h>
+#include <sof/audio/component_ext.h>
+#include <sof/common.h>
+#include <sof/ipc/topology.h>
+#include <sof/lib/wait.h>
+
+#include <sof/sof.h>
+#include <sof/spinlock.h>
+#include <sof/lib_manager.h>
+#include <sof/audio/module_adapter/module/generic.h>
+
+#ifdef LIBRARY_MANAGER
+#include <sof/audio/module_adapter/module/intel_modules.h>
+#endif
+
+#include <errno.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+/* 54cf5598-8b29-11ec-a8a3-0242ac120002 */
+
+DECLARE_SOF_UUID("lib_manager", lib_manager_uuid, 0x54cf5598, 0x8b29, 0x11ec,
+		 0xa8, 0xa3, 0x02, 0x42, 0xac, 0x12, 0x00, 0x02);
+
+DECLARE_TR_CTX(lib_manager_tr, SOF_UUID(lib_manager_uuid), LOG_LEVEL_INFO);
+
+/* TODO: Manifest size should be defined somewhere - to check!!!*/
+#define CAVS18_MAX_MANIFEST_SIZE  (38*1024)
+#define DMA_ELEM_SIZE		32
+#define PAGE_ALIGN			__aligned(HOST_PAGE_SIZE)
```

Comment:
These macros need to go in the correct files or be reused in existing files..

Reply:
I think that they should be in board defines. Upstream process is over I'll check if they can be there.